src/profile-logic/transforms.js

```diff
@@ -1616,66 +1616,51 @@ export function filterCallNodeAndCategoryPathByImplementation(
 }
 
 /**
- * Search through the entire call stack and see if there are any examples of
- * direct recursion.
+ * Search through the entire call node table and see if there are any examples of
+ * direct recursion of funcToCheck.
  */
 export function funcHasDirectRecursiveCall(
```

Comment:
Indeed this function is much simpler when using the derived call node table directly